### PR TITLE
vterrors: addres review comments

### PR DESCRIPTION
**Pull request by sougou**
Addressed review comments. I've changed the error codes for
cases we've agreed on. I've left the more contentious ones
unchanged for now.

BUG=32851872

### CODE REVIEW

**Comment by alainjobart** [comment]
I'll let Michael review this.

File: go/vt/tabletserver/tabletenv/tabletenv.go [hunk 1]
```diff
@@ -29,10 +29,27 @@ var (
 	WaitStats = stats.NewTimings("Waits")
 	// KillStats shows number of connections being killed.
 	KillStats = stats.NewCounters("Kills", "Transactions", "Queries")
-	// InfoErrors shows number of various non critical errors happened.
-	InfoErrors = stats.NewCounters("InfoErrors", "Retry", "DupKey")
 	// ErrorStats shows number of critial erros happened.
-	ErrorStats = stats.NewCounters("Errors", "Fail", "TxPoolFull", "NotInTx", "Deadlock", "Fatal")
+	ErrorStats = stats.NewCounters(
+		"Errors",
```
Comment (michael-berlin) [inline go/vt/tabletserver/tabletenv/tabletenv.go]: What is "Errors"? A total of all values?

In general, it's not advised to have a total in your map as well. Instead, the monitoring system should figure this out for you. Maybe ask Javeen on his opinion on this?

File: go/vt/tabletserver/tabletenv/tabletenv.go [hunk 2]
```diff
@@ -29,10 +29,27 @@ var (
...
+		"OK",
```
Comment (michael-berlin) [inline go/vt/tabletserver/tabletenv/tabletenv.go]: Can you please use the enums here instead?

(I understand that we're probably not going to change the names. But still, this way the connection is more clear where this list comes from :)

File: go/vt/vterrors/aggregate.go [hunk 3]
```diff
@@ -25,10 +25,10 @@ const (
 	PriorityOutOfRange
 	// Potentially retryable errors.
 	PriorityUnavailable
-	PriorityFailedPrecondition
 	PriorityResourceExhausted
```
Comment (michael-berlin) [inline go/vt/vterrors/aggregate.go]: ResourceExhausted should go into the "permanent errors" section.

File: go/vt/vterrors/doc.go [hunk 4]
```diff
@@ -18,8 +18,8 @@ RPCError message that can be used to transmit errors through RPCs, in
 the message payloads. These codes match the names and numbers defined
 by gRPC.
 
-Vitess also defines a VitessError error implementation, that can convert
-any error and add a code to it.
+Vitess also defines a standardized error implementation that allows
+you to build an error with an associated cannonical code.
```
Comment (michael-berlin) [inline go/vt/vterrors/doc.go]: typo: canonical

File: go/vt/tabletserver/tabletenv/tabletenv.go (hunk 1 above)
Comment (sougou) [inline go/vt/tabletserver/tabletenv/tabletenv.go]: NewCounters is actually a misleading constructor. `Errors` is the main tag, and the rest are sub-categories.

File: go/vt/tabletserver/tabletenv/tabletenv.go (hunk 2 above)
Comment (sougou) [inline go/vt/tabletserver/tabletenv/tabletenv.go]: You mean vtrpcpb.Code_OK.String()?

File: go/vt/tabletserver/tabletenv/tabletenv.go (hunk 1 above)
Comment (michael-berlin) [inline go/vt/tabletserver/tabletenv/tabletenv.go]: Ack.

File: go/vt/tabletserver/tabletenv/tabletenv.go (hunk 2 above)
Comment (michael-berlin) [inline go/vt/tabletserver/tabletenv/tabletenv.go]: Yep.

**Comment by sougou** [comment]
LGTM
Last one didn't take.

**Pull request merged**